halo2-base/src/safe_types/tests/var_byte_array.rs

```diff
@@ -0,0 +1,224 @@
+use crate::{
+    gates::{
+        builder::{GateThreadBuilder, RangeCircuitBuilder},
+        RangeChip,
+    },
+    halo2_proofs::{
+        dev::MockProver,
+        halo2curves::bn256::{Bn256, Fr, G1Affine},
+        plonk::{create_proof, keygen_pk, keygen_vk, verify_proof},
+        poly::commitment::ParamsProver,
+        poly::kzg::{
+            commitment::KZGCommitmentScheme, commitment::ParamsKZG, multiopen::ProverSHPLONK,
+            multiopen::VerifierSHPLONK, strategy::SingleStrategy,
+        },
+        transcript::{
+            Blake2bRead, Blake2bWrite, Challenge255, TranscriptReadBuffer, TranscriptWriterBuffer,
+        },
+    },
+    safe_types::SafeTypeChip,
+    utils::testing::base_test,
+    Context,
+};
+use rand::rngs::OsRng;
+use std::vec;
+
+// =========== Utilies ===============
+fn mock_circuit_test<FM: FnMut(&mut Context<Fr>, SafeTypeChip<'_, Fr>)>(mut f: FM) {
+    let mut builder = GateThreadBuilder::mock();
+    let range = RangeChip::default(8);
+    let safe = SafeTypeChip::new(&range);
+    let ctx = builder.main(0);
+    f(ctx, safe);
+    let mut params = builder.config(10, Some(9));
+    params.lookup_bits = Some(3);
```

Comment:
does this work?? `RangeChip` above is using `lookup_bits = 8` but here you're setting it to `3`

If tests are passing I'm very surprised/ confused

Reply:
Good caught. It did pass. I guess it's because we test with `MAX_LEN=4` which is less than 2^3.